src/EditorFeatures/Test/EditAndContinue/EditAndContinueWorkspaceServiceTests.cs

```diff
@@ -1659,6 +1659,44 @@ public async Task Capabilities()
             EndDebuggingSession(debuggingSession);
         }
 
+        [Fact]
+        public async Task Capabilities_SynthesizedNewType()
+        {
+            var source1 = "class C { void M() { } }";
+            var source2 = "class C { void M() { var x = new { Goo = 1 }; } }";
```

Comment:
Does the context of the emitted type matter here? I'm new to this space so may just be obvious. Does this include things like types generated for records?

Reply:
For most of the obvious things we have existing code earlier in the pipeline that will block the right things. For things like records, the type isn't synthesized, just expanded, so we block when the user explicitly adds the definition. For lambdas the type is synthesized, so we still block it when a lambda is added.
This is just blindly blocking _any_ addition without trying to understand why, as a kind of catch all. This is really only for mono which doesn't support type additions, but it will eventually.